src/Sylius/Bundle/SettingsBundle/Controller/SettingsController.php

```diff
@@ -46,7 +46,7 @@ public function showAction(Request $request)
         } catch (MissingOptionsException $e) {
             // When a Settings is not persisted yet, it won't have any initial value in database,
             // so we create a new empty instance.
-            $settings = new Settings([]);
+            $settings = new Settings([], '', null);
```

Comment:
With this way you cannot override the Settings model, since it is hardcoded in the controller. Wouldn't it be better to use the factory here or just let the settings manager make sure you have a proper settings model, and inject the settings factory there?